src/hotspot/share/opto/divnode.cpp

```diff
@@ -570,44 +655,15 @@ const Type* DivINode::Value(PhaseGVN* phase) const {
   const TypeInt *i1 = t1->is_int();
   const TypeInt *i2 = t2->is_int();
   int widen = MAX2(i1->_widen, i2->_widen);
-
-  if( i2->is_con() && i2->get_con() != 0 ) {
-    int32_t d = i2->get_con(); // Divisor
-    jint lo, hi;
-    if( d >= 0 ) {
-      lo = i1->_lo/d;
-      hi = i1->_hi/d;
-    } else {
-      if( d == -1 && i1->_lo == min_jint ) {
-        // 'min_jint/-1' throws arithmetic exception during compilation
-        lo = min_jint;
-        // do not support holes, 'hi' must go to either min_jint or max_jint:
-        // [min_jint, -10]/[-1,-1] ==> [min_jint] UNION [10,max_jint]
-        hi = i1->_hi == min_jint ? min_jint : max_jint;
-      } else {
-        lo = i1->_hi/d;
-        hi = i1->_lo/d;
-      }
+  if (i2->is_con()) {
+    jint d = i2->get_con();    // Divisor
+    if (d == 0) {
+      // this division will always throw an exception
+      return Type::TOP;
```

Comment:
I'd personally prefer to have that code directly below the other cases returning TOP. You can also probably simplify it by testing for `i2 == TypeInt::ZERO` instead.